packages/app/src/vis-packs/core/hooks.ts

```diff
@@ -109,7 +109,19 @@ export function useMappedArrays(
   dims: number[],
   mapping: DimensionMapping,
   autoScale?: boolean
-): [NdArray<NumArray>[], NdArray<NumArray>[]] {
+): [NdArray<NumArray>[], NdArray<NumArray>[]];
+export function useMappedArrays(
+  values: (NumArray | undefined)[],
+  dims: number[],
+  mapping: DimensionMapping,
+  autoScale?: boolean
+): [(NdArray<NumArray> | undefined)[], (NdArray<NumArray> | undefined)[]];
```

Comment:
Overload to be able to pass arrays with `undefined` inside while keeping the old typing when there is no `undefined`.